sklearn/linear_model/ridge.py

```diff
@@ -583,20 +583,24 @@ def fit(self, X, y):
         y : array-like, shape = [n_samples]
             Target values
 
+        sample_weight : float or numpy array of shape (n_samples,)
+            Sample weight.
+
         Returns
         -------
         self : returns an instance of self.
         """
+        if sample_weight is None:
+            sample_weight = 1.
```

Comment:
sample_weight == 1.0 is semantically equivalent to sample_weight.ndim == 0?

On 10 June 2015 at 16:11, Mathieu Blondel <notifications@github.com> wrote:

> In sklearn/linear_model/ridge.py
> <https://github.com/scikit-learn/scikit-learn/pull/4838#discussion_r32090535>
> :
>
> >          Returns
> >          -------
> >          self : returns an instance of self.
> >          """
> > +        if sample_weight is None:
> > +            sample_weight = 1.
>
> Could you also modify this line:
>
> https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/linear_model/ridge.py#L285
> so that has_sw is False if sample_weight == 1.0.
>
> sample_weight is currently implemented with a rescaling of the data. We
> can avoid copying the data in the case sample_weight == 1.0.
>
> Another option is to check for sample_weight == 1.0 directly in
> _rescale_data.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/4838/files#r32090535>.
>